src/Microsoft.DotNet.Build.VstsBuildsApi/VstsBuildHttpClient.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Build.VstsBuildsApi
+{
+    internal class VstsBuildHttpClient : IVstsDefinitionHttpClient
```

Comment:
>VstsBuildHttpClient [](start = 19, length = 19)

Noticeable amount of duplicated logic between this and the release client.  What were the deciding factors in using an interface vs a base class?